src/plugins/intel_gna/frontend/layer_quantizer.hpp

```diff
@@ -254,287 +169,226 @@ inline InferenceEngine::Blob::Ptr fp32_to_precision_blob(InferenceEngine::Blob::
     return result_ptr;
 }
 
-template <class T, class... Args>
-InferenceEngine::Blob::Ptr make_custom_blob(Args&&... args) {
-    return InferenceEngine::make_shared_blob<T>(InferenceEngine::Precision::fromType<T>(), std::forward<Args>(args)...);
-}
-
-template <class T>
-InferenceEngine::Blob::Ptr make_custom_blob(InferenceEngine::Layout layout, InferenceEngine::SizeVector size) {
-    return InferenceEngine::make_shared_blob<T>(
-        InferenceEngine::TensorDesc(InferenceEngine::Precision::fromType<T>(), size, layout));
-}
-
-template<class QuantDesc, class QuantFunc>
-inline void quantizeWeightsBiases(const QuantDesc & quantDesc,
-                                  InferenceEngine::WeightableLayer *wl,
-                                  const QuantFunc &fnc,
-                                  bool isDiagonal = false) {  // for diagonal layer number of weights and biases significatly smaller
-    // for quantized weights
-    auto intWeights =
-        make_custom_blob<typename QuantDesc::WeightsPrecision>(InferenceEngine::C, InferenceEngine::SizeVector({wl->_weights->size()}));
-    intWeights->allocate();
-    if (intWeights->buffer() == nullptr) {
-        IE_THROW(NotAllocated)
-                << "[GNAPlugin] in function " << __PRETTY_FUNCTION__<< ": "
-                << "cannot copy weights for layer :"<< wl->name << " of size" << intWeights->byteSize();
-    }
-
-    int oIdx = wl->outData[0]->getDims().size() - 1;
-    int iIdx = wl->insData[0].lock().get()->getDims().size() - 1;
-
-    auto getBiasSizeForLayer = [&oIdx](InferenceEngine::WeightableLayer *wl) {
-        if (wl->_biases) {
-            return wl->_biases->size();
-        }
+static size_t getBiasSizeForLayer(InferenceEngine::WeightableLayer *wl) {
+    if (wl->_biases) {
+        return wl->_biases->size();
+    } else if (LayerInfo(*wl).isConvolution()) {
+        // calculating biases len using outdata dims: biases number should be equal to output channels number
+        size_t size = InferenceEngine::GetDataDimSize(wl->outData.front(), InferenceEngine::DataDimName::C);
+        return size;
+    } else {
         // calculating biases len using outdata dims
-        auto & dims = wl->outData.front()->getDims();
+        const auto& dims = wl->outData.front()->getDims();
+        const auto& oIdx = wl->outData[0]->getDims().size() - 1;
         return dims[oIdx];
-    };
+    }
+}
 
-    using BiasesPrecision = typename QuantDesc::BiasesPrecision;
-    auto biasMaker = [&] () {
-        InferenceEngine::Blob::Ptr zero;
-        if (!wl->_biases && !shouldAlwaysAllocate<BiasesPrecision>()) {
-            return zero;
+static void getNumRowsColumns(InferenceEngine::WeightableLayer *wl, size_t& num_rows, size_t& num_columns) {
```

Comment:
Detail: Consider change to return std::pair or tuple.
```suggestion
static std::pair<size_t, size_t> getNumRowsColumns(InferenceEngine::WeightableLayer *wl) {
```